### DISCUSSION TITLE
Positron IDE not able to handle multiple interpreter (enviornment) of different versions of Python in one instance,

### DISCUSSION THREAD

**Discussion by Naman-Agrawal19** [posit-dev/positron#4469] [Feedback]
I have two python enviornments mytf and myenv. when i work on mytf, and in between i want to start a new script with myenv, i can't do it, as new script automatically use mytf. When i try to switch the interpreter to myenv manually, `Error: Socket connect timed out after 20 seconds` and not able to work with both the enviornments at the same time. <br> Even if i get successful in changing the env for new script, the env of previous script also change and i lose my variables.<p>
I can do this in VSCode easily, like i can work on two scripts with one on mytf and another on myenv in the same window.<br>
mytf has `python 3.11.9`<br>
myenv has `python 3.12.3` <p> Hope you understand this. Thanks

**Comment by juliasilge** [comment]
You are correct about the current state, yes! Our current take on consoles/environments/windows is that having a 1-to-1 relationship is the best place for us to start for data science users, i.e. one window == one console == one Python env. You can read https://github.com/posit-dev/positron/issues/3173 for details on feature requests to have more than one console.
